atlp-rwanda/commanders-rn-medica
# Ft article implementation 109

#### What does this PR do?
This PR implements the article management and display features on our platform. It includes fetching articles from Supabase, filtering by category, displaying detailed article views, implementing bookmarking and tracking visited articles, sorting by newest and popularity, and displaying trending articles.

#### Description of Task to be completed?
- Implemented API calls to fetch articles from Supabase.
- Created components for displaying all articles with pagination.
- Developed filters for categories to allow seamless filtering.
- Implemented detailed view of single articles with related content.
- Enabled users to bookmark articles and tracked visited articles.
- Added sorting options by newest and popularity metrics.
- Implemented a view for displaying trending articles dynamically.

#### How should this be manually tested?
1. Ensure articles are fetched correctly from Supabase API.
2. Test pagination by loading multiple pages of articles.
3. Use category filters to verify articles are displayed correctly.
4. Check detailed article view for accuracy and related content.
5. Bookmark articles and verify persistence across sessions.
6. Track visited articles and ensure the list updates as expected.
7. Test sorting options by verifying order of articles.
8. Verify the display of trending articles based on engagement metrics.

#### Any background context you want to provide?
This PR builds upon our existing content management system, enhancing the user experience by providing robust article management features. It integrates closely with our backend services and ensures data consistency and reliability.

#### What are the relevant pivotal tracker/Trello stories?
- #37
- #81


#### Questions:
Any feedback or questions regarding the implementation are welcome!
